### PR TITLE
KTOR-9774 Serialize CIO pipelined request bodies

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Client, CIO

**Motivation**
[KTOR-9774](https://youtrack.jetbrains.com/issue/KTOR-9774) CIO can start writing the next pipelined request before the previous request body writer finishes. This allows two request writers to use the same connection concurrently and can corrupt HTTP/1.1 request framing.

**Solution**
Return the asynchronous body writer job from `writeBody` and `writeRequest`, and wait for it in the JVM connection pipeline before advancing to the next request. Response processing and dedicated-connection body writing remain concurrent.

Add a regression test that verifies two non-empty pipelined request bodies are written sequentially and received intact.

### CODE REVIEW

File: ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/ConnectionPipeline.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ internal actual class ConnectionPipeline actual constructor(
                     throw cause
                 }
 
-                writeRequest(task.request, networkOutput, task.context, overProxy, closeChannel = false)
+                writeRequest(task.request, networkOutput, task.context, overProxy, closeChannel = false)?.join()
                 networkOutput.flush()
```
Comment (bjhham) [inline ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/ConnectionPipeline.kt:57]: Maybe we could just use `withContext()` instead of launch / join here to reduce overhead and keep things sequential.

Comment (osipxd) [reply ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/ConnectionPipeline.kt:57]: Do you mean making `writeBody` suspend and replacing its `launch` with `withContext`?
The problem is that `writeBody` is also used outside of pipelining, where it needs to run concurrently with `readResponse`.

**Review by bjhham — APPROVED** [review]
👍
